### PR TITLE
Headless CMS - Strapi

### PR DESCRIPTION
**Pull request by SoniaDumitru**
### Describe the problem being solved:
Created Strapi content for the following pages: 

1. Contact Page: Phone, Email
2. About Page: Board Photos and Bio
3. Volunteer Page: Videos
4. Donate Page: Donate Button Link, Tax Id Number, Phone, Email
5. Header: Social Media Icon Links
6. Footer: Phone, Email
7. Enroll: Email
8. Team: Facebook Link

### Impacted areas in the application: 

List general components of the application that this PR will affect: 
* /frontend/src/components/static_pages/contact.jsx
* /frontend/src/components/static_pages/about.jsx
* /frontend/src/components/donation_page/index.jsx
* /frontend/src/components/static_pages/email.jsx
* /frontend/src/components/static_pages/taxId.jsx
* /frontend/src/components/static_pages/phoneNumber.jsx
* /frontend/src/components/common/header/navbar.jsx
* /frontend/src/components/static_pages/team.jsx

PR checklist
- [ ] I included  a screenshot for FE changes
- [*] I have linked the PR to a Zenhub ticket
- [*] I have checked for merge conflicts
- [ ] I have run the [prettier](https://prettier.io/) command `make pretty`

### CODE REVIEW

File: frontend/src/actions/email_enroll_page.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import axios from "axios";
+
+const getEmail = () => {
+  return dispatch => {
+    const token =
+      "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MSwiaXNBZG1pbiI6dHJ1ZSwiaWF0IjoxNTgxMzgzNDc3LCJleHAiOjE1ODM5NzU0Nzd9.fLn5jTbyPzUMTN-h61DUQtgEdzAXUZMczGqkzFOuwT8";
```
Comment (jsegal-tde) [inline frontend/src/actions/email_enroll_page.js]: I think we should extract this to a function that can be called from all the different places. as this token is used all over the place. 

OR see about using https://www.npmjs.com/package/dotenv to inject the token

File: frontend/src/actions/email_enroll_page.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    const email = "http://localhost:1337/emails";
```
Comment (jsegal-tde) [inline frontend/src/actions/email_enroll_page.js]: Same comment of extracting the `http://localhost:1337` to a function or the `dotenv` package

File: frontend/src/components/SingleCarousel/index.jsx [hunk 3]
```diff
@@ -5,12 +5,23 @@ import logo from "../../assets/logo.png";
...
+      }
+    }
+
     return (
       <Carousel>
         <Carousel.Item>
           <img
             className="d-block w-100 blurImage"
-            src={require("../../assets/" + this.props.image)}
+            // src={require("../../assets/" + this.props.image)}
```
Comment (jsegal-tde) [inline frontend/src/components/SingleCarousel/index.jsx]: please remove comment.

File: frontend/src/components/common/header/navbar.jsx [hunk 4]
```diff
@@ -9,8 +9,37 @@ import {
...
+  state = {
+    facebook: "",
+    twitter: "",
+    instagram: ""
+  };
+
+  componentDidMount() {
+    const token =
+      "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MSwiaXNBZG1pbiI6dHJ1ZSwiaWF0IjoxNTgxMzgzNDc3LCJleHAiOjE1ODM5NzU0Nzd9.fLn5jTbyPzUMTN-h61DUQtgEdzAXUZMczGqkzFOuwT8";
+    const social = "http://localhost:1337/socials";
```
Comment (jsegal-tde) [inline frontend/src/components/common/header/navbar.jsx]: note to extract this

File: frontend/src/components/donation_page/index.jsx [hunk 5]
```diff
@@ -1,8 +1,36 @@
...
 
 class Donate extends Component {
+  state = {
+    paypal: ""
+  };
+
+  componentDidMount() {
+    const token =
+      "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MSwiaXNBZG1pbiI6dHJ1ZSwiaWF0IjoxNTgxMzgzNDc3LCJleHAiOjE1ODM5NzU0Nzd9.fLn5jTbyPzUMTN-h61DUQtgEdzAXUZMczGqkzFOuwT8";
+    const paypal = "http://localhost:1337/donates";
```
Comment (jsegal-tde) [inline frontend/src/components/donation_page/index.jsx]: note to extract this

File: frontend/src/components/enroll/index.jsx [hunk 6]
```diff
@@ -77,4 +84,23 @@ class Enroll extends Component {
...
+    }
+  };
+};
+
+export default connect(
+  mapStateToProps,
+  mapDispatchToProps
+)(Enroll);
+
+// export default Enroll;
```
Comment (jsegal-tde) [inline frontend/src/components/enroll/index.jsx]: please remove comment.

File: frontend/src/components/static_pages/about.jsx [hunk 7]
```diff
@@ -5,8 +5,44 @@ import { Container, Row, Col } from "react-bootstrap";
...
+    aboutShirley: "",
+    kimPhoto: "",
+    pierrePhoto: "",
+    shirleyPhoto: ""
+  };
+
+  componentDidMount() {
+    const token =
+      "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MSwiaXNBZG1pbiI6dHJ1ZSwiaWF0IjoxNTgxMzgzNDc3LCJleHAiOjE1ODM5NzU0Nzd9.fLn5jTbyPzUMTN-h61DUQtgEdzAXUZMczGqkzFOuwT8";
+    const url = "http://localhost:1337/about-pages";
```
Comment (jsegal-tde) [inline frontend/src/components/static_pages/about.jsx]: note to extract this

File: frontend/src/components/static_pages/about.jsx [hunk 8]
```diff
@@ -5,8 +5,44 @@ import { Container, Row, Col } from "react-bootstrap";
...
+        }
+      })
+      .then(response => {
+        this.setState({
+          aboutKim: response.data[0].aboutKim,
+          aboutPierre: response.data[0].aboutPierre,
+          aboutShirley: response.data[0].aboutShirley,
+          kimPhoto: `http://localhost:1337${response.data[0].kimPhoto[0].url}`,
+          pierrePhoto: `http://localhost:1337${response.data[0].pierrePhoto[0].url}`,
+          shirleyPhoto: `http://localhost:1337${response.data[0].shirleyPhoto[0].url}`
```
Comment (jsegal-tde) [inline frontend/src/components/static_pages/about.jsx]: note to extract the `http://localhost:1337`

File: frontend/src/components/static_pages/about.jsx [hunk 9]
```diff
@@ -123,18 +139,7 @@ class About extends Component {
               <h4 className="sub-heading text-left">
```
Comment (jsegal-tde) [inline frontend/src/components/static_pages/about.jsx]: Should we update the <img tag above to use `state` for pierre picture?

File: frontend/src/components/static_pages/about.jsx [hunk 10]
```diff
@@ -153,14 +158,7 @@ class About extends Component {
...
```
Comment (jsegal-tde) [inline frontend/src/components/static_pages/about.jsx]: Should we update the <img tag above to use `state` for Shirley picture?

File: frontend/src/components/static_pages/email.jsx [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+
+class Email extends React.Component {
+  state = {
+    email: ""
+  };
+
+  componentDidMount() {
+    const token =
+      "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MSwiaXNBZG1pbiI6dHJ1ZSwiaWF0IjoxNTgxMzgzNDc3LCJleHAiOjE1ODM5NzU0Nzd9.fLn5jTbyPzUMTN-h61DUQtgEdzAXUZMczGqkzFOuwT8";
+    const email = "http://localhost:1337/emails";
```
Comment (jsegal-tde) [inline frontend/src/components/static_pages/email.jsx]: note to extract this

File: frontend/src/components/static_pages/email.jsx [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+          email: response.data[0].email
+        })
+      )
+      .catch(error => {
+        console.log("An error occurred:", error);
+      });
+  }
+
+  render() {
+    return <>{this.state.email}</>;
```
Comment (jsegal-tde) [inline frontend/src/components/static_pages/email.jsx]: we should make this a `<a href=`mailto:${this.state.email}`>`

File: frontend/src/components/static_pages/phoneNumber.jsx [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+          phone: response.data[0].number
+        })
+      )
+      .catch(error => {
+        console.log("An error occurred:", error);
+      });
+  }
+
+  render() {
+    return <>{this.state.phone}</>;
```
Comment (jsegal-tde) [inline frontend/src/components/static_pages/phoneNumber.jsx]: we should make this a `<a href=`tel:${this.state.phone}`>`

File: frontend/src/components/static_pages/phoneNumber.jsx [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+
+class PhoneNumber extends React.Component {
+  state = {
+    phone: ""
+  };
+
+  componentDidMount() {
+    const token =
+      "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MSwiaXNBZG1pbiI6dHJ1ZSwiaWF0IjoxNTgxMzgzNDc3LCJleHAiOjE1ODM5NzU0Nzd9.fLn5jTbyPzUMTN-h61DUQtgEdzAXUZMczGqkzFOuwT8";
+    const phone = "http://localhost:1337/phone-numbers";
```
Comment (jsegal-tde) [inline frontend/src/components/static_pages/phoneNumber.jsx]: note to extract this

File: frontend/src/components/static_pages/taxId.jsx [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+
+class TaxId extends React.Component {
+  state = {
+    taxId: ""
+  };
+
+  componentDidMount() {
+    const token =
+      "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MSwiaXNBZG1pbiI6dHJ1ZSwiaWF0IjoxNTgxMzgzNDc3LCJleHAiOjE1ODM5NzU0Nzd9.fLn5jTbyPzUMTN-h61DUQtgEdzAXUZMczGqkzFOuwT8";
+    const taxId = "http://localhost:1337/tax-ids";
```
Comment (jsegal-tde) [inline frontend/src/components/static_pages/taxId.jsx]: note to extract this

File: frontend/src/components/static_pages/team.jsx [hunk 16]
```diff
@@ -11,13 +11,35 @@ import Picture6 from "../../assets/action-shots/TTigue Sscott ASayre.jpg";
...
     showImageModal: false,
-    enhancedImage: null
+    enhancedImage: null,
+    facebook: ""
   };
 
+  componentDidMount() {
+    const token =
+      "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MSwiaXNBZG1pbiI6dHJ1ZSwiaWF0IjoxNTgxMzgzNDc3LCJleHAiOjE1ODM5NzU0Nzd9.fLn5jTbyPzUMTN-h61DUQtgEdzAXUZMczGqkzFOuwT8";
+    const social = "http://localhost:1337/socials";
```
Comment (jsegal-tde) [inline frontend/src/components/static_pages/team.jsx]: note to extract this

File: frontend/src/components/volunteer/volunteer.jsx [hunk 17]
```diff
@@ -2,17 +2,43 @@ import React, { Component } from "react";
...
+      video1: "",
+      video2: "",
+      video3: ""
     };
   }
 
+  componentDidMount() {
+    const token =
+      "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MSwiaXNBZG1pbiI6dHJ1ZSwiaWF0IjoxNTgxMzgzNDc3LCJleHAiOjE1ODM5NzU0Nzd9.fLn5jTbyPzUMTN-h61DUQtgEdzAXUZMczGqkzFOuwT8";
+    const videos = "http://localhost:1337/volunteer-videos";
```
Comment (jsegal-tde) [inline frontend/src/components/volunteer/volunteer.jsx]: note to extract this

File: frontend/src/post_strapi.js [hunk 18]
```diff
@@ -0,0 +1,14 @@
+const axios = require("axios");
```
Comment (jsegal-tde) [inline frontend/src/post_strapi.js]: I don't think this file is used and likely can be removed.

File: package.json [hunk 19]
```diff
@@ -12,5 +11,8 @@
     "hooks": {
       "pre-commit": "pretty-quick --staged"
     }
+  },
+  "dependencies": {
+    "axios": "^0.19.2"
```
Comment (jsegal-tde) [inline package.json]: I don't think this needs to be a top level dependency

File: frontend/src/components/donation_page/index.jsx [hunk 20]
```diff
@@ -23,19 +51,22 @@ class Donate extends Component {
...
+            at{" "}
+            <strong>
+              <Email />
+            </strong>
           </p>
 
           <div className="mx-auto donate">
             <form
-              action="https://www.paypal.com/"
+              action={this.state.paypal}
```
Comment (dkaplan88) [inline frontend/src/components/donation_page/index.jsx]: for PayPal -> we are actually going to want the Hosted Button ID Value

File: frontend/src/components/common/header/navbar.jsx [hunk 21]
```diff
@@ -105,4 +105,29 @@ class NavBar extends Component {
...
+  return {
+    gFacebook: () => {
+      dispatch(getStrapi("GET_SOCIAL", "socials"));
+    },
+    gTwitter: () => {
+      dispatch(getStrapi("GET_SOCIAL", "socials"));
+    },
+    gInstagram: () => {
+      dispatch(getStrapi("GET_SOCIAL", "socials"));
+    }
```
Comment (jsegal-tde) [inline frontend/src/components/common/header/navbar.jsx]: You should be able to consolidate these into one call. Something like
```javascript
gSocials: () => {
  dispatch(getStrapi("GET_SOCIAL", "socials"));
}
```

File: frontend/src/components/static_pages/about.jsx [hunk 22]
```diff
@@ -169,4 +139,26 @@ class About extends Component {
...
 
-export default About;
+const mapStateToProps = state => {
+  return {
+    aboutKim: state.strapi.aboutKim,
+    aboutPierre: state.strapi.aboutPierre,
+    aboutShirley: state.strapi.aboutShirley,
+    kimPhoto: `http://localhost:1337${state.strapi.kimPhoto}`,
+    pierrePhoto: `http://localhost:1337${state.strapi.pierrePhoto}`,
+    shirleyPhoto: `http://localhost:1337${state.strapi.shirleyPhoto}`
```
Comment (jsegal-tde) [inline frontend/src/components/static_pages/about.jsx]: I believe you will need a forward slash `/` in between localhost:1337 and the state variables.

File: frontend/src/components/volunteer/volunteer.jsx [hunk 23]
```diff
@@ -127,5 +129,29 @@ class Volunteer extends Component {
...
+  return {
+    gVideo1: () => {
+      dispatch(getStrapi("GET_SOCIAL", "videos"));
+    },
+    gVideo2: () => {
+      dispatch(getStrapi("GET_SOCIAL", "videos"));
+    },
+    gVideo3: () => {
+      dispatch(getStrapi("GET_SOCIAL", "videos"));
+    }
```
Comment (jsegal-tde) [inline frontend/src/components/volunteer/volunteer.jsx]: these can also be consolidated to one mapping

```javascript
gVideo: () => {
  dispatch(getStrapi("GET_SOCIAL", "videos"));
}
```

File: frontend/src/components/common/header/navbar.jsx [hunk 24]
```diff
@@ -9,9 +9,18 @@ import {
...
 
 class NavBar extends Component {
+  componentDidMount() {
+    this.props.gFacebook();
+    this.props.gTwitter();
+    this.props.gInstagram();
+  }
+
   render() {
+    console.log(this.props);
```
Comment (jsegal-tde) [inline frontend/src/components/common/header/navbar.jsx]: please remove

File: frontend/src/components/donation_page/index.jsx [hunk 25]
```diff
@@ -69,4 +81,23 @@ class Donate extends Component {
     );
   }
 }
-export default Donate;
+const mapStateToProps = state => {
+  console.log("link:", state.strapi.donateLink);
```
Comment (jsegal-tde) [inline frontend/src/components/donation_page/index.jsx]: please remove

File: frontend/src/components/static_pages/about.jsx [hunk 26]
```diff
@@ -5,9 +5,17 @@ import { Container, Row, Col } from "react-bootstrap";
...
+import { connect } from "react-redux";
+import getStrapi from "../../actions/strapi.js";
 
 class About extends Component {
+  componentDidMount() {
+    this.props.gAbout();
+  }
+
   render() {
+    console.log(this.props.kimPhoto);
```
Comment (jsegal-tde) [inline frontend/src/components/static_pages/about.jsx]: please remove

File: frontend/src/actions/email_enroll_page.js [hunk 27]
```diff
@@ -0,0 +1,18 @@
+import axios from "axios";
+
+const getEmail = () => {
```
Comment (dkaplan88) [inline frontend/src/actions/email_enroll_page.js]: I think this file can be deleted now?

File: frontend/src/reducers/strapi.js [hunk 28]
```diff
@@ -0,0 +1,83 @@
...
+  sponsor9: Sponsor9,
+  sponsor10: Sponsor10,
+  sponsor11: Sponsor11,
+  sponsor12: Sponsor12
+};
+
+export default function getStrapi(state = initialState, action) {
+  switch (action.type) {
+    case "GET_EMAIL":
+      return { ...state, email: action.response.data[0].email };
```
Comment (dkaplan88) [inline frontend/src/reducers/strapi.js]: I do not think we are not going to be able to do `.data[0]`

I believe we are always going to want to take the last one for all of these? (most recently created) 

so maybe grab [last](https://lodash.com/docs/4.17.15#last) from lodash so we can do 

```
 let lastEmail = last(action.response.data)'
 if (lastEmail) {
   return { ...state, email: lastEmail.email };
}
```

File: frontend/src/reducers/strapi.js [hunk 29]
```diff
@@ -0,0 +1,83 @@
...
+  taxId: "82-1904373",
+  donateLink: "https://www.paypal.com/",
+  token: "",
+  facebook: "https://www.facebook.com/theymim/",
+  instagram: "https://www.instagram.com/theyoungmasterbuilders/",
+  twitter: "https://twitter.com/YMIMtweets",
+  video1: "https://www.youtube.com/embed/PK9ESRMGq74",
+  video2: "https://www.youtube.com/embed/jdsqht1m1rE",
+  video3: "https://www.youtube.com/embed/GwXt3tL6FqY",
+  sponsor1: Sponsor1,
```
Comment (dkaplan88) [inline frontend/src/reducers/strapi.js]: this looks good in `GET_SPONSORS` action

but for initialState this is going to have to be `sponsors: [Sponsor1, Sponsor2, ...]` to work properly

**Comment by jsegal-tde** [comment]
Closing as most of these changes were used in https://github.com/the-difference-engine/ymim/pull/434 which has been ✅ and deployed to staging

**Pull request closed without merging**